src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(health = 100, name) {
+    this.health = health;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  checkHealth() {
+    if (this.health <= 0) {
+      this.index = Animal.alive.indexOf(this);
+      Animal.alive.splice(this.index, 1);
+    }
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(health = 100, name, hidden) {
+    super(health, name);
+    this.hidden = hidden;
+    this.hidden = false;
+    this.health = 100;
```

Comment:
The `health` property is already set by the `Animal` constructor with a default value of `100`. There is no need to reassign it to `100` in the `Herbivore` constructor.